File: javascript/insertionSort/insertionSort.js
"use strict"

function swap(arr, a, b) {
	const tmp = arr[a];
	arr[a] = arr[b];
	arr[b] = tmp;
	return arr;
}

export default function insertionSort(arr) {
	let i = 1;
	while(i < arr.length) {
		let j = i;
		while(j > 0 && arr[j-1] > arr[j]) {
			swap(arr, j, j-1);
			j--;
		}
		i++;
	}
	return arr;
}

function main() {
	let toSort = [];
	for(const a of Deno.args) {
		toSort.push(a);	
	}
	if(toSort.length == 0) return;
	console.log(insertionSort(toSort));
}

main();

File: java/insertionSort/app/src/main/java/insertionsort/insertionSort.java
package javaAlgorithms;

import java.util.Arrays;


class insertionSort {
	public static void swap(int[] array, int a, int b) {
		int tmp = array[a];
		array[a] = array[b];
		array[b] = tmp;
	}

	public static void insertionSortInt(int[] array) {
		int i = 1;
		while(i < array.length) {
			int j = i;
			while(j > 0 && array[j-1] > array[j]) {
				swap(array, j, j-1);
				j--;
			}
			i++;
		}
	}

	public static void main(String[] args) {
		int[] a = {3, 2, 1, 0};
		insertionSortInt(a);
		System.out.println(Arrays.toString(a));
	}
}
